pandas/tests/indexes/common.py

```diff
@@ -400,7 +429,9 @@ def test_equals(self, index):
 
         assert index.equals(index)
         assert index.equals(index.copy())
-        assert index.equals(index.astype(object))
+        if not (type(index) is Index and not isinstance(index.dtype, np.dtype)):
+            # doesn't hold for e.g. IntegerDtype
+            assert index.equals(index.astype(object))
```

Comment:
i think there's a GH issue about how strict .equals should be.  backward-compat notwithstanding, I would like the Index versions to be appreciably stricter (since they are mainly used to see "can we fastpath this method?" where a non-strict .equals defeats the purpose), closer to the EA versions (in particular datetimelike).

My recollection is that in most cases, we do test for `obj.equals(obj.astype(object))`.  if im wrong and we don't have precedent for that, then i agree it'd be better not to start doing so.

Unless you're suggesting the opposite?  im pre-caffeine.

Reply:
The issue is https://github.com/pandas-dev/pandas/issues/33940

I am personally not sure whether I like the strict behaviour of EA / Series or the loose behaviour of Index (will comment on the issue again), but so the question is here: do we let Index[EA] for now follow the documented Index.equals behaviour or do we make this more strict? 

(if we would deprecate the loose Index.equals behaviour, it might make sense to already keep it strict here, but so that depends on https://github.com/pandas-dev/pandas/issues/33940)

Reply:
could also make a strict `_equals`